packages/plugins/typescript/react-apollo/tests/react-apollo.spec.ts

```diff
@@ -997,6 +997,39 @@ export function useListenToCommentsSubscription(baseOptions?: ApolloReactHooks.S
       expect(content.content).not.toContain(`export type TestSubscriptionResult = ApolloReactCommon.SubscriptionResult<TestSubscription>;`);
       await validateTypeScript(content, schema, docs, {});
     });
+    it('should generate lazy query hooks', async () => {
+      const documents = parse(/* GraphQL */ `
+        query feed {
+          feed {
+            id
+            commentCount
+            repository {
+              full_name
+              html_url
+              owner {
+                avatar_url
+              }
+            }
+          }
+        }
+      `);
+      const docs = [{ filePath: '', content: documents }];
+
+      const content = (await plugin(
+        schema,
+        docs,
+        { withHooks: true, withComponent: false, withHOC: false },
+        {
+          outputFile: 'graphql.tsx',
+        }
+      )) as Types.ComplexPluginOutput;
+
+      expect(content.content).toBeSimilarStringTo(`
+  export function useFeedLazyQuery(baseOptions?: ApolloReactHooks.LazyQueryHookOptions<FeedQuery, FeedQueryVariables>) {
```

Comment:
Wouldn't a name like `useLazyFeedQuery` be more correct?

Reply:
We decided to keep it as-is for now, I think both forms are correct